zerver/tornado/event_queue.py

```diff
@@ -523,6 +525,9 @@ def fetch_events(query: Mapping[str, Any]) -> Dict[str, Any]:
             if user_profile_id != client.user_profile_id:
                 raise JsonableError(_("You are not authorized to get events from this queue"))
             client.event_queue.prune(last_event_id)
+            if last_event_id != client.event_queue.newest_pruned_id:
+                logging.warning("'last_event_id' %s was not in this queue" % (last_event_id,))
+                raise JsonableError(_("'last_event_id' %s was not in this queue") % (last_event_id,))
```

Comment:
I think we should still check for the two different situations, and give different error messages for the "already garbage-collected" case vs. "was never added to the the queue" case.

Reply:
\*shrug\* Added.